examples/step-57/doc/results.dox

```diff
@@ -0,0 +1,376 @@
+
+<table align="center" border="1">
+  
+  <tr>
+    <th>&nbsp;</th>
+    <th colspan="2">Mesh0</th>
+    <th colspan="2">Mesh1</th>
+    <th colspan="2">Mesh2</th>
+    <th colspan="2">Mesh3</th>
+    <th colspan="2">Mesh4</th>
+  </tr>
+
+  <tr>
+    <th>Newton's iter </th>
+    <th>Residual      </th>
+    <th>FGMRES        </th>
+    <th>Residual      </th>
+    <th>FGMRES        </th>
+    <th>Residual      </th>
+    <th>FGMRES        </th>
+    <th>Residual      </th>
+    <th>FGMRES        </th>
+    <th>Residual      </th>
+    <th>FGMRES        </th>
+  </tr>
+
+  <tr>
+    <th>1         </th>
+    <th>7.40396e-3</th>
+    <th>3         </th>
+    <th>1.05562e-3  </th>
+    <th>3         </th>
+    <th>4.94796e-4  </th>
+    <th>3         </th>
+    <th>2.5624e-4 </th>
+    <th>2         </th>
+    <th>1.26733e-4  </th>
+    <th>2         </th>
+  </tr>
+
+  <tr>
+    <th>2         </th>
+    <th>3.86766e-3  </th>
+    <th>4         </th>
+    <th>1.3549e-5  </th>
+    <th>3         </th>
+    <th>1.41981e-6  </th>
+    <th>3        </th>
+    <th>1.29108e-6  </th>
+    <th>4         </th>
+    <th>6.14794e-7  </th>
+    <th>4         </th>
+  </tr>
+
+  <tr>
+    <th>3         </th>
+    <th>1.60421e-3</th>
+    <th>4         </th>
+    <th>1.24836e-9  </th>
+    <th>3         </th>
+    <th>9.11557e-11  </th>
+    <th>3         </th>
+    <th>3.35933e-11 </th>
+    <th>3         </th>
+    <th>5.86734e-11 </th>
+    <th>2         </th>
+  </tr>
+
+  <tr>
+    <th>4         </th>
+    <th>9.26748e-4  </th>
+    <th>4         </th>
+    <th>2.75537e-14  </th>
+    <th>4         </th>
+    <th>1.39986e-14 </th>
+    <th>5         </th>
+    <th>2.18864e-14 </th>
+    <th>5         </th>
+    <th>3.38787e-14 </th>
+    <th>5         </th>
+  </tr>
+
+  <tr>
+    <th>5          </th>
+    <th>1.34601e-5</th>
+    <th>4          </th>
+    <th>&nbsp;    </th>
+    <th>&nbsp;    </th>
+    <th>&nbsp;    </th>
+    <th>&nbsp;    </th>
+    <th>&nbsp;    </th>
+    <th>&nbsp;    </th>
+    <th>&nbsp;    </th>
+    <th>&nbsp;    </th>
+  </tr>
+
+  <tr>
+    <th>6         </th>
+    <th>2.5235e-8    </th>
+    <th>5    </th>
+    <th>&nbsp;    </th>
+    <th>&nbsp;    </th>
+    <th>&nbsp;    </th>
+    <th>&nbsp;    </th>
+    <th>&nbsp;    </th>
+    <th>&nbsp;    </th>
+    <th>&nbsp;    </th>
+    <th>&nbsp;    </th>
+  </tr>
+
+  <tr>
+    <th>7         </th>
+    <th>1.38899e-12    </th>
+    <th>4    </th>
+    <th>&nbsp;    </th>
+    <th>&nbsp;    </th>
+    <th>&nbsp;    </th>
+    <th>&nbsp;    </th>
+    <th>&nbsp;    </th>
+    <th>&nbsp;    </th>
+    <th>&nbsp;    </th>
+    <th>&nbsp;    </th>
+  </tr>
+
+  <tr>
+    <th>8           </th>
+    <th>4.68224e-15 </th>
+    <th>4           </th>
+    <th>&nbsp;    </th>
+    <th>&nbsp;    </th>
+    <th>&nbsp;      </th>
+    <th>&nbsp;      </th>
+    <th>&nbsp;      </th>
+    <th>&nbsp;      </th>
+    <th>&nbsp;      </th>
+    <th>&nbsp;      </th>
+  </tr>
+</table>
```

Comment:
I must be missing something here: The table seems to indicate that you get quadratic convergence. But don't you just assemble the matrix in the first step of the nonlinear iteration? I'm surprised this is enough for such a good convergence.